File: leetcode/0-300/51-100/66.Plus One.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int carry = 1;
        int i = digits.size() - 1;
        while (carry != 0 && i >= 0) {
            digits[i] += carry;
            carry = 0;
            if (digits[i] > 9) {
                carry = digits[i] % 9;
                digits[i] = 0;
            } else break;
            --i;
        }
        
        if (carry > 0) {
            digits.insert(digits.begin(), carry);
        }
        
        return digits;
    }
};

File: leetcode/0-300/51-100/66.Plus One.rs
//https://leetcode.com/problems/plus-one

impl Solution {
    pub fn plus_one(digits: Vec<i32>) -> Vec<i32> {
        let mut carry = 1;
        let mut plus_one = digits.iter().rev().map(|n| {
            let num = n + carry;
            if num > 9 {
                carry = num % 9
            } else {
                carry = 0
            }
            num % 10
        }).collect::<Vec<i32>>();

        plus_one.reverse();

        if carry > 0 {
            plus_one.insert(0, carry);
        }

        plus_one
    }
}